### ISSUE TITLE
get_edge() does not work with node instances as imput

### ISSUE THREAD

**Issue by Welgriv** [pydot/pydot#272]
```
import pydot

g = pydot.Dot()
n1 = pydot.Node('n1')
n2 = pydot.Node('n2')
g.add_node(n1)
g.add_node(n2)
g.add_edge(pydot.Edge(src=n1, dst=n2))

print(g.get_edge(n1, n2))
```

return an empty list.

**Comment by inahus99** [comment]
Opened #573 for this.

The cause is that `Edge` reduces its endpoints to names on construction and `add_node` keys nodes by name, but the lookups never applied the same reduction  so the objects never matched their own keys. `del_node()` and `del_edge()` already normalised their arguments this way, so the PR just brings the getters in line and moves the shared bit into one helper.

The snippet in the report returns the edge now, and `get_node(n1)` works too.

@lkk7 no rush on this one — flagging it since it's a small, self-contained fix that mirrors the normalisation `del_node()`/`del_edge()` already do.

**Comment by ferdnyc** [comment]
@inahus99 

My only reticence on this is the following:

When using the `get_*()` lookups with objects, it's reasonable to expect that what's returned will be only the graph element that involves _those specific objects_. But, none of our `get_*()` lookups actually work like that, they're purely name-based, and I'm not sure it would meet expectations to return what appears to be more than the lookup asked for.

For example, `get_node()` doesn't accept `Node` objects as input, because you already _have_ the `Node` object you appear to be asking for, and having it retrieve other objects that share the same name is probably more confusing than useful:

```python
>>> import pydot
>>> g = pydot.Graph("G")
>>> n1 = pydot.Node("N", color="red")
>>> n2 = pydot.Node("N", color="blue")
>>> g.add_node(n1)
>>> g.add_node(n2)
>>> [n.to_string() for n in g.get_node(n1.get_name())]
['N [color=red];', 'N [color=blue];']
```

If that call were `g.get_node(n1)`, you'd probably expect _only_ the node corresponding to `n1` to be returned, but that's not what would happen.

By the same token, with `.get_edge()`:
```python
>>> import pydot
>>> g = pydot.Graph("G")
>>> n1 = pydot.Node("a")
>>> n2 = pydot.Node("b")
>>> e1 = pydot.Edge(n1, n2)
>>> e2 = pydot.Edge("a", "b", dir="both")
>>> g.add_edge(e1)
>>> g.add_edge(e2)
>>> [e.to_string() for e in g.get_edge(n1.get_name(), n2.get_name())]
['a -> b;', 'a -> b [dir=both];']
```

If a user was to call `g.get_edge(n1, n2)`, would they expect to get back _both_ of those edges, or only the `e1` that was created directly from `n1` and `n2`?

**Comment by ferdnyc** [comment]
(Personally, I recommend keeping references to any graph elements you create, so that you never need to use `pydot.Graph.get_*()`.)

The main use case for `pydot.Graph.get_*()` is element retrieval in graphs that _**weren't**_ created programmatically, like a parsed graph definition:

```python
>>> graph_src = """
... graph G {
...  a -- b [color=red];
...  a -- b [color=blue];
... }"""
>>> g = pydot.graph_from_dot_data(graph_src)[0]
>>> [e.to_string() for e in g.get_edge("a", "b")]
['a -- b [color=red];', 'a -- b [color=blue];']
```

...But if you didn't create the graph using the API, then you don't have existing `pydot.Node("a")` and `pydot.Node("b")` objects to pass to `.get_edge()` anyway, so it's hard to see where the actual use case for `pydot.Graph.get_edge(pydot.Node, pydot.Node)` really is.

**Comment by ferdnyc** [comment]
(But I'm willing to be persuaded that `pydot.Graph.get_edge(pydot.Node, pydot.Node)` is a vital piece of API that has practical real-world applications and isn't confusing. I don't see it, but anyone who believes that is welcome to make their case.)

**Comment by lkk7** [comment]
Yes, I never thought about that, but both node and edge cases would be very confusing for the user.
Not possible with the current shape of the codebase.
